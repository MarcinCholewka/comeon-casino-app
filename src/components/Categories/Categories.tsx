import { Grid, Header, List, Loader } from 'semantic-ui-react';
import { toast } from 'react-toastify';
import { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';

import { BASE_API } from '@utils/constants';
import { CategoryItem } from '@components/CategoryItem';

export type TCategory = {
  id: number;
  name: string;
};

export const Categories = () => {
  const { t } = useTranslation('translation', { keyPrefix: 'Categories' });
  const [categories, setCategories] = useState<TCategory[]>([]);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const fetchCategories = async () => {
      try {
        setIsLoading(true);

        const response = await fetch(`${BASE_API}/categories`);
        const data = await response.json();

        setCategories(data);
      } catch (error: unknown) {
        console.log(
          'ðŸš€ ~ file: Categories.tsx:25 ~ fetchCategories ~ error:',
          error,
        );

        toast.error('Something went wrong! Please try again.');
      } finally {
        setIsLoading(false);
      }
    };

    fetchCategories();
  }, []);

  return (
    <Grid.Column width='four'>
      <Header as='h3' dividing={true}>
        {t('header')}
      </Header>
      {isLoading ? (
        <Loader active={true} inline='centered' size='big' />
      ) : (
        <List selection={true} animated={true}>
          {categories.map((category) => (
            <CategoryItem key={category.id} category={category} />
          ))}
        </List>
      )}
    </Grid.Column>
  );
};
