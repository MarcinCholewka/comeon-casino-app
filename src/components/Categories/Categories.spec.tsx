import { render } from '@testing-library/react';
import { Categories } from './Categories';

describe('<Categories/ >', () => {
  test('should render without crashing', () => {
    const { container, getByText } = render(<Categories />);

    expect(getByText('Categories')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
