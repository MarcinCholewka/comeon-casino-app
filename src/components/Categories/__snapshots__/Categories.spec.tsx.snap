// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories/ > should render without crashing 1`] = `
<div>
  <div
    class="four wide column"
  >
    <h3
      class="ui dividing header"
    >
      Categories
    </h3>
    <div
      class="ui animated selection list"
      role="list"
    />
  </div>
</div>
`;
